### PR TITLE
feat: add hustler details/gear page version: 1

### PR DESCRIPTION
**Pull request by cmarouan**
This pull request contains the first version of the hustler details + gear using Dope tabs:

Remarks:

- The respect field not returned at the moment from the backend
- Hustler's Tier color in Gears needs to be fixed on the backend side after that the color will be displayed in the frontend (already implemented in the front)

Todo:

- Dope tabs mobile style
- Dope tabs should be integrated with the header
- The actions should be added to the buttons (at the moment only the display)

### CODE REVIEW

File: packages/web/src/features/hustlers/components/HustlerInformation/actions/SellHustler.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Button } from '@chakra-ui/react';
+
+export default function SellHustler() {
+  return (
+    <Button type="button" variant="primary" loadingText="Processing..." w="100%">
```
Comment (smakosh) [inline packages/web/src/features/hustlers/components/HustlerInformation/actions/SellHustler.tsx:5]: This should accept onClick prop

File: packages/web/src/features/hustlers/components/HustlerInformation/actions/index.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import SellHustler from './SellHustler';
```
Comment (smakosh) [inline packages/web/src/features/hustlers/components/HustlerInformation/actions/index.tsx:1]: Can we stick with absolute imports?

File: packages/web/src/features/hustlers/components/HustlerInformation/actions/index.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import styled from '@emotion/styled';
+
+export const ActionsContainer = styled.div`
+  padding: 18px;
+  width: 100%;
+`;
+
+export default function index() {
```
Comment (smakosh) [inline packages/web/src/features/hustlers/components/HustlerInformation/actions/index.tsx:9]: ES6 arrow function

**Review by smakosh — CHANGES_REQUESTED** [review]
Left some notes, will add more once the first ones are fixed.

Also let's server side render this page using getServerSideProps Next js API.

File: packages/web/src/components/hustler/ZoomControls.tsx [hunk 4]
```diff
@@ -31,6 +31,7 @@ const ZoomControls = ({ config, setHustlerConfig }: ConfigureHustlerProps) => {
       isVehicle,
       renderName: renderName,
     });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (cmarouan) [reply packages/web/src/components/hustler/ZoomControls.tsx]: done

File: packages/web/src/features/hustlers/components/HustlerInformation/HustlerDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      </ControlsBody>
+      <PanelFooter>
+        <div />
+        <Button type="button" loadingText="Processing...">
+          Change Appearance
+        </Button>
+      </PanelFooter>
+    </ControlsWrapper>
+  );
+}
```
[the remark this answers is not in the document]
Comment (cmarouan) [reply packages/web/src/features/hustlers/components/HustlerInformation/HustlerDetails.tsx]: done

File: packages/web/src/features/hustlers/components/HustlerInformation/HustlerDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    value: title,
+  },
+  {
+    key: 'Respect:',
+    value: respect,
+  },
+];
+
+export default function HustlerDetails(porps: HustlerDetailsProps) {
+  const [details, _] = useState(formatHustlerDetails(porps));
```
[the remark this answers is not in the document]
Comment (cmarouan) [reply packages/web/src/features/hustlers/components/HustlerInformation/HustlerDetails.tsx]: done

File: packages/web/src/features/hustlers/components/HustlerInformation/HustlerDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+      <ControlsBody>
+        {details.map(detail => (
+          <LineContent key={detail.key}>
+            <LineKey>{detail.key}</LineKey>
+            <LineValue>{detail.value}</LineValue>
+          </LineContent>
+        ))}
+      </ControlsBody>
+      <PanelFooter>
+        <div />
```
[the remark this answers is not in the document]
Comment (cmarouan) [reply packages/web/src/features/hustlers/components/HustlerInformation/HustlerDetails.tsx]: 👀

File: packages/web/src/features/hustlers/components/HustlerInformation/HustlerInformationTabs.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+};
+
+export type HustlerDetailsProps = {
+  name: string;
+  id: string;
+  title: string;
+  respect: string;
+};
+
+export default function HustlerInformationTabs({ huslter }: HustlerInformationTabsProps) {
```
[the remark this answers is not in the document]
Comment (cmarouan) [reply packages/web/src/features/hustlers/components/HustlerInformation/HustlerInformationTabs.tsx]: done

File: packages/web/src/features/hustlers/components/HustlerInformation/actions/SellHustler.tsx [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { Button } from '@chakra-ui/react';
+
+export default function SellHustler() {
```
[the remark this answers is not in the document]
Comment (cmarouan) [reply packages/web/src/features/hustlers/components/HustlerInformation/actions/SellHustler.tsx]: done

File: packages/web/src/features/hustlers/components/HustlerInformation/actions/SellHustler.tsx (hunk 1 above)
Comment (cmarouan) [reply packages/web/src/features/hustlers/components/HustlerInformation/actions/SellHustler.tsx]: done

File: packages/web/src/features/hustlers/components/HustlerInformation/actions/index.tsx (hunk 2 above)
Comment (cmarouan) [reply packages/web/src/features/hustlers/components/HustlerInformation/actions/index.tsx]: sure !

File: packages/web/src/features/hustlers/components/HustlerInformation/actions/index.tsx (hunk 3 above)
Comment (cmarouan) [reply packages/web/src/features/hustlers/components/HustlerInformation/actions/index.tsx]: :D

File: packages/web/src/features/hustlers/components/HustlerInformation/HustlerGear.tsx [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+            </LineValue>
+          </LineContent>
+        ))}
+      </ControlsBody>
+      <PanelFooter>
+        <Button type="button" loadingText="Processing...">
+          Cancel
+        </Button>
+
+        <Button type="button" variant="primary" loadingText="Processing...">
```
[the remark this answers is not in the document]
Comment (cmarouan) [reply packages/web/src/features/hustlers/components/HustlerInformation/HustlerGear.tsx]: The form will be added once the actions are integrated for sure! thanks for the remark

**Review by smakosh — APPROVED** [review]
LGTM

**Comment by subimage** [comment]
Closing PR because code did not match design. Already working on this in another branch.

**Pull request closed without merging**